src/ci-stats/client.ts

```diff
@@ -1,53 +1,109 @@
-      jenkinsUrl: process.env.BUILDKITE_BUILD_URL,
-      prId: process.env.GITHUB_PR_NUMBER || null,
+  async createBuild() {
+    return await this.req<CiStatsBuild>({
+      url: '/v1/build',
+      method: 'POST',
+      data: {
+        jenkinsJobName: process.env.BUILDKITE_PIPELINE_NAME,
+        jenkinsJobId: process.env.BUILDKITE_BUILD_ID,
+        jenkinsUrl: process.env.BUILDKITE_BUILD_URL,
+        prId: process.env.GITHUB_PR_NUMBER || null,
+      },
     });
+  }
 
-    return resp.data as CiStatsBuild;
-  };
-
-  addGitInfo = (buildId: string) => {
-    return this.http.post(`/v1/git_info?buildId=${buildId}`, {
-      branch: (process.env.BUILDKITE_BRANCH || '').replace(/^(refs\/heads\/|origin\/)/, ''),
-      commit: process.env.BUILDKITE_COMMIT,
-      targetBranch:
-        process.env.GITHUB_PR_TARGET_BRANCH || process.env.BUILDKITE_PULL_REQUEST_BASE_BRANCH || null,
-      mergeBase: process.env.GITHUB_PR_MERGE_BASE || null, // TODO confirm GITHUB_PR_MERGE_BASE or switch to final var
+  async addGitInfo(buildId: string) {
+    await this.req({
+      url: `/v1/git_info?buildId=${buildId}`,
+      method: 'POST',
+      data: {
+        branch: (process.env.BUILDKITE_BRANCH || '').replace(/^(refs\/heads\/|origin\/)/, ''),
+        commit: process.env.BUILDKITE_COMMIT,
+        targetBranch:
+          process.env.GITHUB_PR_TARGET_BRANCH || process.env.BUILDKITE_PULL_REQUEST_BASE_BRANCH || null,
+        mergeBase: process.env.GITHUB_PR_MERGE_BASE || null, // TODO confirm GITHUB_PR_MERGE_BASE or switch to final var
+      },
     });
-  };
+  }
 
-  completeBuild = (buildStatus: string, buildId = process.env.CI_STATS_BUILD_ID) => {
-    return this.http.post(`/v1/build/_complete?id=${buildId}`, {
-      result: buildStatus,
+  async completeBuild(buildStatus: string, buildId = process.env.CI_STATS_BUILD_ID) {
+    await this.req({
+      url: `/v1/build/_complete?id=${buildId}`,
+      method: 'POST',
+      data: {
+        result: buildStatus,
+      },
     });
-  };
+  }
+
+  private async req<R>(options: AxiosRequestConfig) {
+    let attempt = 0;
+
+    // eslint-disable-next-line no-constant-condition
+    attemptRequest: while (true) {
+      attempt += 1;
+
+      try {
+        const resp = await this.http.request<R>(options);
+        return resp.data;
+      } catch (error) {
+        const status: number | undefined = error?.response?.status;
+
+        if (attempt < MAX_ATTEMPTS && (typeof status === 'undefined' || status >= 500)) {
+          const delay = attempt * this.baseRetryMs;
+          this.console.error(
+            `${status} response from ${this.baseUrl}, retrying in`,
+            delay / 1000,
+            'seconds:',
+            'toJSON' in error && typeof error.toJSON === 'function' ? error.toJSON() : error,
+          );
+          await new Promise((resolve) => setTimeout(resolve, delay));
+          continue attemptRequest;
+        }
+
+        throw error;
```

Comment:
Could you just wrap the `throw error` in an `else{}` and get rid of the continue/label? Or you could invert the condition in the `if` and put the throw in it instead